lib/msf/core/db_manager/import/qualys/scan.rb

```diff
@@ -58,22 +59,25 @@ def import_qualys_scan_xml(args={}, &block)
       end
 
       # VULNS are confirmed, PRACTICES are unconfirmed vulnerabilities
-      host.elements.each('VULNS/CAT | PRACTICES/CAT') do |cat|
-        port = cat.attributes['port']
-        protocol = cat.attributes['protocol']
-        cat.elements.each('VULN | PRACTICE') do |vuln|
+      host.xpath('VULNS/CAT | PRACTICES/CAT').each do |cat|
+        port = cat.attr('port')
+        protocol = cat.attr('protocol')
+        cat.xpath('VULN | PRACTICE').each do |vuln|
           refs = []
-          qid = vuln.attributes['number']
-          severity = vuln.attributes['severity']
-          title = vuln.elements['TITLE'].text.to_s
-          vuln.elements.each('VENDOR_REFERENCE_LIST/VENDOR_REFERENCE') do |ref|
-            refs.push(ref.elements['ID'].text.to_s)
+          qid = vuln.attr('number')
+          severity = vuln.attr('severity')
+          title = vuln.xpath('TITLE').first&.text
+          vuln.xpath('VENDOR_REFERENCE_LIST/VENDOR_REFERENCE').each do |ref|
+            id = ref.xpath('ID').first&.text
+            refs.push(id) if id
           end
-          vuln.elements.each('CVE_ID_LIST/CVE_ID') do |ref|
-            refs.push('CVE-' + /C..-([0-9\-]{9,})/.match(ref.elements['ID'].text.to_s)[1])
+          vuln.xpath('CVE_ID_LIST/CVE_ID').each do |ref|
+            id = ref.xpath("ID").first&.text
+            refs.push("CVE-#{id}") if id
```

Comment:
See issue above on CVE IDs, you'll have the same issue where with appending an extra `CVE-` in front of the CVE value.